### PR TITLE
fix Gamma distribution's NaN gradients for zero inputs

### PR DESCRIPTION
**Pull request by PascalIversen**
*Issue #, if available:*
fix for #1066

*Description of changes:*
Added masking to prevent NaN gradients of the `Gamma` for `x=0`.
This is important for fitting mixture distributions such as: 
```python
mdo = MixtureDistributionOutput([GaussianOutput(), GammaOutput()]).
```
Example code:
```python
import mxnet as mx
from gluonts.mx.distribution import Gamma

a =mx.nd.array([1])
b =mx.nd.array([1])

a.attach_grad()
b.attach_grad()

with mx.autograd.record():
    loss = mx.nd.mean(Gamma(a,b).loss(x=mx.nd.array([0])))
    loss.backward()
 

print(a.grad)
print(b.grad)
```
will return
```python
[0.]
<NDArray 1 @cpu(0)>

[0.]
<NDArray 1 @cpu(0)>
```

instead of
```python
[nan]
<NDArray 1 @cpu(0)>

[0.]
<NDArray 1 @cpu(0)>

```

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by PascalIversen** [comment]
I added a test case which fails on master but passes on this branch.

**Review by lostella — APPROVED** [review]
